frontend/Rhyolite/Frontend/Widget.hs

```diff
@@ -78,3 +82,25 @@ extensibleListWidgetWithSize n x0 xs0 addAtEnd itemWidget = do
           valuesMapD = joinDynThroughMap $ fmap (fmap snd) $ resultMapD
           valuesD = fmap Map.elems valuesMapD
   return valuesD
+
+-- | Widget used as a modal div for widgets that want to take some action when clicked anywhere but itself, such as dropdown widgets or the like.
+  -- The first argument is a CSS class name, the suggested CSS class styling for use of this widget is as follows:
+  -- position: fixed;
+  -- top: 0;
+  -- bottom: 0;
+  -- right: 0;
+  -- left: 0;
+  -- z-index: 100;
```

Comment:
`z-index` only works for explicitly positioned elements. So the code here works but must be placed before all other DOM-building. As such it is far less convenient to use. Here's an example of the same idea, but where we introduce a monad to make it much more ergonomic (you can use `tellModal` from anywhere in the app and it will build the DOM at the bottom to give it natural stacking order on top of everything else).

https://gitlab.com/obsidian.systems/tezos-bake-monitor/blob/develop/tezos-bake-central/frontend/src/Frontend/Modal/Base.hs#L106-108

Also note in this example, we pass the close Event into the modal-building function so that it can choose to accept background clicks or not accept them as close events. It also accepts Esc key.

Reply:
I'm not sure that's true, given the presence of `position: fixed`

Reply:
Ah you're probably right for the commented example. I suppose the point is, if you don't pass the right CSS here it won't work as intended. There is a way to make it work regardless of the CSS however, and that's to use it at the bottom of the DOM. But the API is far too painful to use as-is when you do it that way, hence the example.

Reply:
At the very least I'd recommend we name the function less generally. It over promises with this name (more like `withBackdropWhenConfiguredCorrectly`). Even with explicitly positioned elements, I believe it will only work within that particular "stack" (the nuances escape me), so your backdrop still might not cover the whole page.